gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValueUtils.java

```diff
@@ -613,6 +614,79 @@ private static float floatValue(PropertyValue value, int type) {
     }
   }
 
+  /**
+   * Date and DateTime utilities.
+   */
+  public static class Date {
+    /**
+     * Compares two time property values and returns the chronological first one.
+     * <p>
+     * Note that the comparison of a {@link java.time.LocalDate} with a {@link LocalDateTime}
+     * will cause casting the {@link java.time.LocalDate} to a {@link java.time.LocalDateTime}
+     * instance, but just for the comparison. The return value is not casted.
+     *
+     * @param a first value
+     * @param b second value
+     *
+     * @return the time property value that is the chronological first
+     */
+    public static PropertyValue min(PropertyValue a, PropertyValue b) {
+      return isLessOrEqualThan(a, b) ? a : b;
+    }
+
+    /**
+     * Compares two time property values and returns the chronological last one.
+     * <p>
+     * Note that the comparison of a {@link java.time.LocalDate} with a {@link LocalDateTime}
+     * will cause casting the {@link java.time.LocalDate} to a {@link java.time.LocalDateTime}
+     * instance, but just for the comparison. The return value is not casted.
+     *
+     * @param a first value
+     * @param b second value
+     *
+     * @return the time property value that is the chronological last
+     */
+    public static PropertyValue max(PropertyValue a, PropertyValue b) {
+      return isLessOrEqualThan(a, b) ? b : a;
+    }
+
+    /**
+     * Compares two date or datetime property values and returns true, if the first one is earlier or equal to
+     * the second one. Note that the comparison of a {@link java.time.LocalDate} with a {@link LocalDateTime}
+     * will cause casting the {@link java.time.LocalDate} to a {@link java.time.LocalDateTime} instance.
+     *
+     * @param aValue first value
+     * @param bValue second value
+     *
+     * @return true, iff the first time is earlier or equal to the second one
+     */
+    private static boolean isLessOrEqualThan(PropertyValue aValue, PropertyValue bValue) {
```

Comment:
I think it would be better to return the compare result directly and call this function `compare`, since you are using `compareTo()` anyway.

Reply:
I also thought about that but then I decided to make it equal to the Numeric class. But now you convinced me to do it as I once planned :)